alternator/executor.cc

```diff
@@ -1130,15 +1130,8 @@ static future<executor::request_return_type> create_table_on_shard0(tracing::tra
         // This should never happen, the ID is supposed to be unique
         co_return api_error::internal(format("Table with ID {} already exists", schema->id()));
     }
-    db::schema_tables::add_table_or_view_to_schema_mutation(schema, ts, true, schema_mutations);
-    // we must call before_create_column_family callbacks - which allow
-    // listeners to modify our schema_mutations. For example, CDC may add
-    // another table (the CDC log table) to the same keyspace.
-    // Unfortunately the convention is that this callback must be run in
-    // a Seastar thread.
-    co_await seastar::async([&] {
-        mm.get_notifier().before_create_column_family(*schema, schema_mutations, ts);
-    });
+    auto table_mutations = co_await service::prepare_new_column_family_announcement(sp, *ksm, schema, ts);
+    std::move(table_mutations.begin(), table_mutations.end(), std::back_inserter(schema_mutations));
```

Comment:
I see you had to use this back_inserter idiom in several places, which suggests that maybe the `prepare_new_column_family_annoucement()` function, instead of returning a new vector, should append mutations to a given vector? It will make the different uses simpler?

Just like the old function "add_table_or_view_to_schema_mutation" which added something to an existing vector, and so do the notifiers.

Reply:
That's a good idea, I'll do it. After all, the new `prepare_new_column_family_annoucement` has been introduced to be called together with `prepare_new_keyspace_announcement`. So, it makes sense for this function to behave differently from the other `prepare_` functions that usually return all mutations for an announcement.